File: src/main/java/org/moinex/app/config/FlywayConfig.java
/*
 * Filename: FlywayConfig.java
 * Created on: March  5, 2025
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.app.config;

import javax.sql.DataSource;
import org.flywaydb.core.Flyway;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class FlywayConfig {
    @Bean
    public Flyway flyway(DataSource dataSource) {
        Flyway flyway =
                Flyway.configure()
                        .dataSource(dataSource)
                        .baselineOnMigrate(true)
                        .mixed(true)
                        .load();
        flyway.migrate();
        return flyway;
    }
}

File: src/main/kotlin/org/moinex/config/FlywayConfig.kt
/*
 * Filename: FlywayConfig.kt (original filename: FlywayConfig.java)
 * Created on: March  5, 2025
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrated to Kotlin on 18/03/2026
 */

package org.moinex.config

import org.flywaydb.core.Flyway
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.sql.DataSource

@Configuration
class FlywayConfig {
    @Bean
    fun flyway(dataSource: DataSource): Flyway {
        val flyway =
            Flyway
                .configure()
                .dataSource(dataSource)
                .baselineOnMigrate(true)
                .mixed(true)
                .load()
        flyway.migrate()
        return flyway
    }
}
